File: Sources/UsageWidget/Domain/ProviderID.swift
import Foundation

/// Stable identifier for a usage provider. Open by design — adding a provider is just
/// a new static constant plus a concrete `UsageProvider` implementation.
struct ProviderID: RawRepresentable, Hashable, Codable, Sendable {
    let rawValue: String
    init(rawValue: String) { self.rawValue = rawValue }

    static let claude = ProviderID(rawValue: "claude")
    static let codex = ProviderID(rawValue: "codex")
}

File: windows/src/Core/Domain/ProviderId.cs
namespace LLMUsageWidget.Core.Domain;

/// <summary>Stable identifier for a usage provider. Open by design — adding a provider is just a
/// new constant plus a concrete provider implementation.</summary>
public readonly record struct ProviderId(string RawValue)
{
    public static readonly ProviderId Claude = new("claude");
    public static readonly ProviderId Codex = new("codex");
    public static readonly ProviderId Copilot = new("copilot");

    public override string ToString() => RawValue;
}
